JSIL.Libraries/Sources/JSIL.Core.Types.js

```diff
@@ -16,7 +16,7 @@ JSIL.ImplementExternals("System.Object", function ($) {
   $.RawMethod(true, "CheckType",
     function (value) {
       var type = typeof (value); 
-      return (type=== "object" || type === "number" || type === "string");
```

Comment:
Probably we should not return true for `undefined`. Also we should implement Object methods for raw JS object for it.